src/System.Net.Http/src/System/Net/Http/SocketsHttpHandler/ContentLengthReadStream.cs

```diff
@@ -133,6 +133,32 @@ private void Finish()
                 _connection.ReturnConnectionToPool();
                 _connection = null;
             }
+
+            public override async Task DrainAsync(int maxDrainBytes)
+            {
+                Debug.Assert(_connection != null);
+                Debug.Assert(_contentBytesRemaining > 0);
+
+                if (_contentBytesRemaining > (ulong)maxDrainBytes)
+                {
+                    return;
+                }
+
+                while (true)
+                {
+                    if (_contentBytesRemaining <= (ulong)_connection.RemainingBuffer.Length)
+                    {
+                        _connection.ConsumeFromRemainingBuffer((int)_contentBytesRemaining);
+                        Finish();
+                        break;
+                    }
+
+                    _contentBytesRemaining -= (ulong)_connection.RemainingBuffer.Length;
+                    _connection.ConsumeFromRemainingBuffer(_connection.RemainingBuffer.Length);
+
+                    await _connection.FillAsync();
```

Comment:
ConfigureAwait(false)

Reply:
I don't know why I seem to have a mental block about this, lol.